src/main/java/com/slicingdice/jslicer/exceptions/api/SlicingDiceException.java

```diff
@@ -1,7 +1,31 @@
 package com.slicingdice.jslicer.exceptions.api;
 
+import org.json.JSONObject;
+
 public class SlicingDiceException extends RuntimeException {
-	public SlicingDiceException(String message){
+
+	private int code;
+	private String message;
+	private Object moreInfo;
+
+	public SlicingDiceException(final String message){
 		super(message);
 	}
+
+	public SlicingDiceException(final JSONObject data){
+		super(data.getString("message"));
+
+		this.code = data.getInt("code");
+		this.message = data.getString("message");
+		this.moreInfo = data.get("more-info");
+	}
+
+	@Override
+	public String toString() {
+		return "SlicingDiceException{" +
```

Comment:
The Codacy showed issues about the indentation and format (break line) of this code.